### PR TITLE
Add support for Porkbun

### PR DESCRIPTION
**Pull request by sbell8-chwy**
Add support for [porkbun](https://porkbun.com).

Porkbun works very similar to GoDaddy, with the exception that auth is handled by passing the api key and secret as part of the post body.

I haven't actually run this against my account yet. Will do that this weekend before merging should happen.

### CODE REVIEW

**Comment by nickbabcock** [comment]
Very cool, one you run it against your account, let me know and then I'll verify too.

**Comment by sbell8-chwy** [comment]
Ran this against my account both when no changes were needed and when a change was needed.
Needed a couple tweaks, but looks good now.

**Review by nickbabcock — COMMENTED** [review]
Looking good, just a few comments.

File: README.md [hunk 1]
```diff
@@ -293,6 +294,33 @@ password = "IpUpdatePassword"
...
+
+# This is the api key, you can create one here:
+# https://porkbun.com/account/api
+key = "abc123"
+
+# The password for the key, top secret! Only visible once when you create the key.
+secret = "ef"
+
+# The records to update. "" = "example.com", "a" = "a.example.com"
+records = [ "", "a" ]
```
[the remark this answers is not in the document]
Comment (sbell8-chwy) [reply README.md:315]: I understand the desire for consistency. Given that porkbun doesn't use @ for root domains, and there are several places matching is done, I felt it safest to allow the user to enter "@" in the config, and then map it to "" immediately in the porkbun module. The way this works the user will be free to use either "" or "@" for root domain, and I added that to the README

Let me know what you think.

**Comment by sbell8-chwy** [comment]
Sorry for the slow response. Got a bit busy. Those all look like good points, updated all.

**Review by nickbabcock — APPROVED** [review]
Looks good, thanks for the contribution. I will cut a new release after merging this.

**Pull request merged**